src/functional/ffront/past_to_itir.py

```diff
@@ -226,20 +280,25 @@ def _compute_field_slice(node: past.Subscript):
         return out_field_slice_
 
     def _visit_stencil_call_out_arg(
-        self, node: past.Expr, **kwargs
+        self, node_out: past.Expr, node_field_domain: past.Dict, **kwargs
     ) -> tuple[itir.SymRef, itir.FunCall]:
-        if isinstance(node, past.Subscript):
+        if isinstance(node_out, past.Subscript):
```

Comment:
if `node_out` is a Subscript, we probably must not have the `field_domain` argument (otherwise it's overspecified (or we need to define what the slicing means relative to the given domain)). Let's go with either domain or slicing. Please add a test in `test_past_parsing` that checks for an error in this case, i.e. `field_op(..., out=out_field[...], domain=...`. (`...` meaning, please fill in what makes sense.)

Reply:
Note that there is another Subscript below inside `elif isinstance(node_out, past.TupleExpr)`. Test both cases.